src/middlewared/middlewared/plugins/vm/devices/cdrom.py

```diff
@@ -36,3 +36,47 @@ def xml_linux(self, *args, **kwargs):
                 ]
             }
         )
+
+    def _validate(self, device, verrors, old=None, vm_instance=None, update=True):
+        path = device['attributes']['path']
+        if not os.path.exists(path):
+            verrors.add('attributes.path', f'Unable to locate CDROM device at {path!r}')
+        elif not self.middleware.call_sync('vm.device.disk_uniqueness_integrity_check', device, vm_instance):
+            verrors.add(
+                'attributes.path',
+                f'{vm_instance["name"]} has "{self.identity()}" already configured'
+            )
+
+        if not verrors:
+            # We would like to check now if libvirt will actually be able to read the iso file
+            # How this works is that if libvirt user is not able to read the file, libvirt automatically changes
+            # ownership of the iso file to the libvirt user so that it is able to read however there are cases where
+            # even this can fail with perms like 000 or maybe parent path(s) not allowing access.
+            # To mitigate this, we can do the following:
+            # 1) See if owner of the file is libvirt user
+            # 2) If it's not libvirt user:
+            # a) Check if libvirt user can access the file
+            # b) Change ownership of the file to libvirt user as libvirt would eventually do
+            # 3) Check if libvirt user can access the file
+            libvirt_user = self.middleware.call_sync(
+                'user.query', [['username', '=', LIBVIRT_USER]], {'get': True}
+            )
+            libvirt_group = self.middleware.call_sync('group.query', [['group', '=', LIBVIRT_USER]], {'get': True})
+            current_owner = os.stat(path)
+            is_valid = False
+            if current_owner.st_uid != libvirt_user['uid']:
+                if self.middleware.call_sync('filesystem.can_access_as_user', LIBVIRT_USER, path, {'read': True}):
+                    is_valid = True
+                else:
+                    os.chown(path, libvirt_user['uid'], libvirt_group['gid'])
```

Comment:
This will fail if the ISO is on a read-only filesystem

Reply:
@themylogin currently the logic has been copied over as is, is it okay if we tackle this in a separate PR ?

Reply:
ok